### PR TITLE
[Zero-Dim] support input 0D Tensor for maximum,minimum,allclose,sigmoid_focal_loss

### PR DESCRIPTION
**Pull request by FlyingQianMM**
### PR types

New features
### PR changes

APIs
### Describe

[Zero-Dim] support input 0D Tensor for some binary api.
---
```
paddle.maximum
paddle.minimum
paddle.allclose
paddle.nn.functional.sigmoid_focal_loss
```

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/xpu/test_zero_dim_tensor_xpu.py [hunk 1]
```diff
@@ -703,6 +705,26 @@ def test_argsort(self):
...
         self.assertEqual(x2.grad.numpy(), 0)
 
+    def test_sigmoid_focal_loss(self):
+        logit = paddle.to_tensor(
+            [[0.97, 0.91, 0.03], [0.55, 0.43, 0.71]], dtype='float32'
+        )
+        label = paddle.to_tensor(
+            [[1.0, 0.0, 0.0], [0.0, 1.0, 0.0]], dtype='float32'
+        )
+        one = paddle.to_tensor([1.0], dtype='float32')
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/unittests/xpu/test_zero_dim_tensor_xpu.py]: 这个可以删了吧

File: python/paddle/fluid/tests/unittests/xpu/test_zero_dim_tensor_xpu.py [hunk 2]
```diff
@@ -703,6 +705,26 @@ def test_argsort(self):
...
+        fg_num_0 = paddle.full([], 2.0)
+        fg_num_1 = paddle.full([1], 2.0)
+        np.testing.assert_array_equal(
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/unittests/xpu/test_zero_dim_tensor_xpu.py:723]: 需要测下反向

File: python/paddle/fluid/tests/unittests/xpu/test_zero_dim_tensor_xpu.py (hunk 1 above)
Comment (FlyingQianMM) [reply python/paddle/fluid/tests/unittests/xpu/test_zero_dim_tensor_xpu.py]: 已删除，感谢~

File: python/paddle/fluid/tests/unittests/xpu/test_zero_dim_tensor_xpu.py (hunk 2 above)
Comment (FlyingQianMM) [reply python/paddle/fluid/tests/unittests/xpu/test_zero_dim_tensor_xpu.py:723]: 已添加反向梯度测试，感谢~

**Review by zhwesky2010 — DISMISSED** [review]
LGTM

**Review by jzhang533 — DISMISSED** [review]
LGTM

**Review by jzhang533 — APPROVED** [review]

**Review by zhwesky2010 — APPROVED** [review]
LGTM
